### PR TITLE
18447 Fix opening of preview areas on clicking preview button

### PR DESCRIPTION
**Pull request by myabc**
https://community.openproject.org/work_packages/18447

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://fix-18447-preview-button-open-preview.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://fix-18447-preview-button-open-preview.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by myabc** [comment]
https://community.openproject.org/work_packages/18447

**Comment by myabc** [comment]
@floriank Not sure if you still wanted to take a look at this. As-is this _works for me_, but Cucumber features are currently failing and I've yet to find time to debug them.

I did a minimal port of jQuery code to Angular. Further scope for refactoring:
- switching to Angular `$http`, `$anchorScroll` services (not tested though): https://gist.github.com/myabc/7506ad8b27eb35a6a019
- creating a `PreviewService` or such like.
- removing the hard-coded target element (`#preview`) and making as second `previewArea` directive that registers itself as being interested in previews for a particular form (not sure how best to handle intra-directive communication here).

**Comment by floriank** [comment]
I can verify the functionality - it seems to work fine. 

I'll have a look at the cucumber tests.

**Comment by myabc** [comment]
@floriank thanks!

**Comment by floriank** [comment]
@myabc @ulferts It seems that the PR actually broke the issue update view, as the angular directive `has-preview` is not properly compiled due to the fact that the element is created after the dom has been loaded. 

I'll be fixing it. Dunno how yet - I am open for suggestions, though.

**Comment by floriank** [comment]
This PR will also solve https://community.openproject.org/work_packages/18451

File: app/assets/javascripts/application.js.erb [hunk 1]
```diff
@@ -1213,23 +1213,3 @@ var SubmitConfirm = (function($) {
...
-                                        scrollTop: $('#preview').offset().top
-                                      },
-                                      'slow');
-                                  }
-        });
-
-        return false;
-      });
-    });
-})(jQuery);
```
Comment (ulferts) [inline app/assets/javascripts/application.js.erb:1235]: :+1: for removing code from the application.js

**Pull request merged**